Feature/154621250/reviews

kensanni opened the pull request:
#### What does this PR do?
>- This PR allows users to rate and review shops and products

#### Description of Task to be completed?
>- Ratings and Review
#### How should this be manually tested?
>- git clone this project @https://github.com/andela/blade-runner-rc.git
>- run meteor npm install to install all dependencies
>- run "reaction "to start the server
>- server would be up and running on localhost:3000

#### Any background context you want to provide?
>- N/A

#### What are the relevant pivotal tracker stories?
>- https://www.pivotaltracker.com/n/projects/2137297/stories/153741761

#### Screenshots (if appropriate):
<img width="1440" alt="screen shot 2018-02-06 at 3 14 34 pm" src="https://user-images.githubusercontent.com/22421283/35864511-4c8d1b0a-0b52-11e8-947a-3a4b2e61530e.png">
<img width="1428" alt="screen shot 2018-02-06 at 3 15 10 pm" src="https://user-images.githubusercontent.com/22421283/35864513-4cae5a0e-0b52-11e8-9d4c-0c515286fb85.png">
<img width="1440" alt="screen shot 2018-02-06 at 3 16 10 pm" src="https://user-images.githubusercontent.com/22421283/35864515-4cd1e06e-0b52-11e8-99ab-72a5cfc5ea72.png">
<img width="1423" alt="screen shot 2018-02-06 at 3 17 14 pm" src="https://user-images.githubusercontent.com/22421283/35864517-4cf65570-0b52-11e8-8950-d720f584b26a.png">


#### Questions:
>- N/A

Commit:
feature(reviews-and-ratings): add ratings and reviews to the application

create schema for product reviews
install react stars component
display reviews on product details page
add create reviews server methods
[Starts #154621250]

kensanni committed:
feature(component): create shop review and rating component

kensanni committed:
feature(shop-rating): Implement shop rating and review

chinezenwosu commented on imports/plugins/included/default-theme/client/styles/shopReview/shopReview.less:
```diff
@@ -0,0 +1,16 @@
+.button-margin {
+  margin-top: 30px;
+  
```
Remove extra space

chinezenwosu commented on imports/plugins/included/product-detail-simple/client/components/productReview.js:
```diff
@@ -0,0 +1,130 @@
+                    <h4 className="media-heading" style={{ marginBottom: "12px" }}>
+                      <ReactStars onChange={rating => { this.setState({ rating }); }} count={5} size={18} value={this.state.rating} />
+                    </h4>
+                    <textarea style={{ border: "2px solid #5cde86", boxShadow: "none", marginBottom: "4px" }} placeholder="Leave a review ..." cols="2" rows="2"
```
Move inline style to css file

chinezenwosu replied:
Move inline style to css file

chinezenwosu commented on imports/plugins/included/shop-reviews/client/components/shopReview.js:
```diff
@@ -0,0 +1,139 @@
+    });
+  }
+  render() {
+    const { reviews } = this.props;
```
Abstract this into a method

chinezenwosu commented on imports/plugins/included/shop-reviews/server/init.js:
```diff
@@ -0,0 +1,21 @@
+import { Reaction, Logger, Hooks } from "/server/api";
+import { Shops } from "/lib/collections";
+
+function addRolesToVisitors() {
```
Consider changing the word 'Visitors'?...

kensanni committed:
feature(feedback): Implement TTl feedback

[#154621250]

kensanni committed:
feature(feedback): Implement TTl feedback

[#154621250]

kensanni committed:
feature(shop-rating): fix merge conflict

kensanni committed:
feature(circleCI): fix circleCI errors

chinezenwosu commented on imports/plugins/included/product-detail-simple/client/components/productReview.js:
```diff
@@ -0,0 +1,128 @@
+                      size={40}
+                      className={"img-responsive mt-2"}
+                      email={this.state.user.emails[0].address}
+                      name={this.state.user.name === "" ? this.state.user.emails[0].address : this.state.user.name}
```
Destructure user and review at the beginning of render.

chinezenwosu commented on imports/plugins/included/shop-reviews/client/components/shopReview.js:
```diff
@@ -0,0 +1,132 @@
+                      <ReactionAvatar
+                        size={40}
+                        className={"img-responsive mt-2"}
+                        email={this.state.user.emails[0].address}
```
Destructure here too

Commit:
feature(reviews-and-ratings):

reduce space between user avatar and review text box
display stars under product and shop ratings
refactor state to destructured parameters
reposition product rating to be after product description
[#154621250]

Commit:
Merge branch 'feature/154621250/reviews' of https://github.com/andela/blade-runner-rc into feature/154621250/reviews
Merge github remote

Commit:
Merge branch 'develop' into feature/154621250/reviews

Commit:
feature(reviews):

fix eslint error in shopReview.js
